SSA/Projects/RISCV64/Semantics.lean

```diff
@@ -25,23 +32,87 @@ def UTYPE_pure64_AUIPC (imm : BitVec 20) (pc : BitVec 64) : BitVec 64 :=
   BitVec.add (BitVec.signExtend 64 (BitVec.append imm (0x000 : BitVec 12))) pc
 
 def SHIFTIWOP_pure64_RISCV_SLLIW (shamt : BitVec 5) (rs1_val : BitVec 64) : BitVec 64 :=
-  BitVec.signExtend 64 ((BitVec.extractLsb' 0 32 rs1_val) <<< shamt)
+  BitVec.signExtend 64 (BitVec.shiftLeft (BitVec.extractLsb' 0 32 rs1_val) (shamt).toNat)
+
+def SHIFTIWOP_pure64_RISCV_SLLIW_bv (shamt : BitVec 5) (rs1_val : BitVec 64) : BitVec 64 :=
+  BitVec.signExtend 64 ((BitVec.extractLsb' 0 32 rs1_val)  <<< shamt)
+
+theorem SHIFTIWOP_pure64_RISCV_SLLIW_eq_SHIFTIWOP_pure64_RISCV_SLLIW_bv (shamt : BitVec 5) (rs1_val : BitVec 64) :
+  SHIFTIWOP_pure64_RISCV_SLLIW shamt rs1_val = SHIFTIWOP_pure64_RISCV_SLLIW_bv shamt rs1_val :=
+  by
+  unfold SHIFTIWOP_pure64_RISCV_SLLIW SHIFTIWOP_pure64_RISCV_SLLIW_bv
+  rfl
```

Comment:
```suggestion
theorem SHIFTIWOP_pure64_RISCV_SLLIW_eq_SHIFTIWOP_pure64_RISCV_SLLIW_bv (shamt : BitVec 5) (rs1_val : BitVec 64) :
    SHIFTIWOP_pure64_RISCV_SLLIW shamt rs1_val = SHIFTIWOP_pure64_RISCV_SLLIW_bv shamt rs1_val := by
  unfold SHIFTIWOP_pure64_RISCV_SLLIW SHIFTIWOP_pure64_RISCV_SLLIW_bv
  rfl
```

- theorem statement lines are typically indented by four spaces
- `by` typically follows right after the `:=`
Similarly, below.